sync/src/chain.rs

```diff
@@ -209,6 +212,8 @@ pub struct ChainSync {
 	max_download_ahead_blocks: usize,
 	/// Network ID
 	network_id: U256,
+	/// Transactions Queue
+	transaction_queue: Mutex<TransactionQueue>,
```

Comment:
Transaction queue is not multithreaded. Methods `import` and `remove` internally would have to lock everythng anyway. I'm not sure if we would gain anything from making it thread safe.

What are the most expensive operations? Do you think that instead of removing / adding right away it should just maintain a thread pool internaly?